src/Type/Symfony/CacheInterfaceGetDynamicReturnTypeExtension.php

```diff
@@ -0,0 +1,45 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Symfony;
+
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Type\ClosureType;
+use PHPStan\Type\DynamicMethodReturnTypeExtension;
+use PHPStan\Type\GeneralizePrecision;
+use PHPStan\Type\Type;
+
+final class CacheInterfaceGetDynamicReturnTypeExtension implements DynamicMethodReturnTypeExtension
+{
+
+	public function getClass(): string
+	{
+		return 'Symfony\Contracts\Cache\CacheInterface';
+	}
+
+	public function isMethodSupported(MethodReflection $methodReflection): bool
+	{
+		return $methodReflection->getName() === 'get';
+	}
+
+	public function getTypeFromMethodCall(MethodReflection $methodReflection, MethodCall $methodCall, Scope $scope): ?Type
+	{
+		if (!isset($methodCall->getArgs()[1])) {
+			return null;
+		}
+
+		$callbackReturnType = $scope->getType($methodCall->getArgs()[1]->value);
+		if ($callbackReturnType instanceof ClosureType) {
```

Comment:
CacheInterface accepts any callable, `instanceof *Type` is always wrong.

`Type` has `getCallableParametersAcceptors`, check `isCallable()->yes()` and then call this method.